### DISCUSSION TITLE
Is there any way to use the GPU of the AMD 5700G APU for accelerating inference?

### DISCUSSION THREAD

**Discussion by xoich** [oobabooga/text-generation-webui#4377] [Q&A]
The 5700G is a an APU, with good integrated graphics. As far as I am aware it's not supported by ROCm. So apparently there is nothing I can do with the GPU?

**Comment by berkut1** [comment]
Even if it were possible, you wouldn't gain anything. because the APU does not have VRAM, RAM is too slow for AI.

**Reply by xoich** [reply]
No gain compared to CPU seems unlikely, the CPU inference itself is using RAM.

**Reply by berkut1** [reply]
There is a clear example where the new Nvidia driver uses RAM instead of VRAM. The speed can drop to 10 times and is almost equal to the CPU speed. Example https://github.com/oobabooga/text-generation-webui/issues/4304#issuecomment-1771342925, but you can find even more similar examples, the problem with Nvidia drivers exist since May.

**Reply by xoich** [reply]
Could it be though that the APU's GPU has a better connection to RAM than a dGPU? I'm not an expert, but it might be that the communication between dGPU and RAM has to pass through the CPU.

**Reply by berkut1** [reply]
https://github.com/LostRuins/koboldcpp/releases/tag/v1.47.2 you can try this, it has many options for GPUs (OpenBlas)
P.S modern GPU use direct memory access (DMA) to access the ram directly without the CPU intervention.
